internal/repositories/rewards.go

```diff
@@ -0,0 +1,218 @@
+package repositories
+
+import (
+	"context"
+	"fmt"
+
+	gmodel "github.com/DIMO-Network/identity-api/graph/model"
+	"github.com/DIMO-Network/identity-api/internal/helpers"
+	"github.com/DIMO-Network/identity-api/models"
+	"github.com/DIMO-Network/shared/dbtypes"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/volatiletech/sqlboiler/v4/queries/qm"
+	"github.com/volatiletech/sqlboiler/v4/types"
+	"golang.org/x/exp/slices"
+)
+
+type RewardsCursor struct {
+	Week      int
+	VehicleID int
+}
+
+func RewardToAPI(reward models.Reward) gmodel.Earning {
+	stEarn := dbtypes.NullDecimalToInt(reward.StreakEarnings)
+	adEarn := dbtypes.NullDecimalToInt(reward.AftermarketEarnings)
+	syEarn := dbtypes.NullDecimalToInt(reward.SyntheticEarnings)
+
+	return gmodel.Earning{
+		Week:                    reward.IssuanceWeek,
+		Beneficiary:             common.BytesToAddress(reward.ReceivedByAddress.Bytes),
+		ConnectionStreak:        reward.ConnectionStreak.Ptr(),
+		StreakTokens:            stEarn,
+		AftermarketDeviceID:     reward.AftermarketTokenID.Ptr(),
+		AftermarketDeviceTokens: adEarn,
+		SyntheticDeviceID:       reward.SyntheticTokenID.Ptr(),
+		SyntheticDeviceTokens:   syEarn,
+		SentAt:                  reward.EarnedAt,
+		VehicleID:               reward.VehicleID,
+	}
+}
+
+func (r *Repository) GetEarningsByVehicleID(ctx context.Context, tokenID int) (*gmodel.VehicleEarnings, error) {
+	pHelp := helpers.PaginationHelper[RewardsCursor]{}
+
+	type rewardStats struct {
+		TokenSum   types.NullDecimal `boil:"token_sum"`
+		TotalCount int               `boil:"total_count"`
+	}
+	var stats rewardStats
+
+	limit := new(int)
+	*limit = 100
+
+	err := models.Rewards(
+		qm.Select(
+			fmt.Sprintf(
+				`sum(%s + %s + %s) as token_sum`,
```

Comment:
Do you know what happens if there are no rows? We said the total would be `BigInt!`, so it needs to be there. Just checking, since we treat this as `types.NullDecimal`.

Reply:
It returns `nil`

Reply:
Did you get a chance to check?